File: src/ProjetoCanil/Modelo/Vacinavel.cs
namespace ProjetoCanil.Modelo;

/// <summary>
/// Define o contrato para animais que podem receber vacinas.
/// </summary>
public interface IVacinavel
{
    /// <summary>
    /// Registra a aplicação de uma vacina.
    /// </summary>
    /// <param name="nomeVacina">Nome da vacina aplicada.</param>
    void AplicarVacina(string nomeVacina);

    /// <summary>
    /// Lista todas as vacinas aplicadas no animal.
    /// </summary>
    /// <returns>Lista de nomes das vacinas.</returns>
    List<string> ListarVacinas();
}

File: legacy-java/modelo/Vacinavel.java
package Bpoo.projeto_canil.modelo;

import java.util.List;

/**
 * Define o contrato para animais que podem receber vacinas.
 */
public interface Vacinavel {

    /**
     * Registra a aplicação de uma vacina.
     * @param nomeVacina nome da vacina aplicada.
     */
    void aplicarVacina(String nomeVacina);

    /**
     * Lista todas as vacinas aplicadas no animal.
     * @return lista de nomes das vacinas.
     */
    List<String> listarVacinas();
}
